Utilites/Tokenizer/tokenize-ps3.ps1

```diff
@@ -107,11 +113,12 @@ try {
         try {
             if ($allVars.ContainsKey($matchedItem)) {
                 $variableValue = $allVars[$matchedItem].Value
-                Write-Verbose "Found custom variable '$matchedItem' in build or release definition" 
+                Write-Verbose "Found custom variable '$matchedItem' in build or release definition with value '$variableValue'"
             } else {
-                if ($Configuration.$environmentName.CustomVariables.$matchedItem) {
-                    $variableValue = $Configuration.$environmentName.CustomVariables.$matchedItem
-                    Write-Verbose "Found variable '$matchedItem' in configuration with value '$variableValue" 
+                $environmentVariableValue = $environmentNames | foreach { $Configuration.$_.CustomVariables.$matchedItem } | where { $_ } | select -Last 1
```

Comment:
Can you add a comment here which says that select -Last 1 is the logic which selects the right value since all the overriding environments should be later in the $environmentNames list than default env.